### PR TITLE
[PM-6302, PM-6303] Add duo state and connector message on browser/desktop

### PR DESCRIPTION
**Pull request by jlf0dev**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

This PR has two purposes.
The first is the client side changes of using the `state` parameter of the duo authentication process. This ensures session validity during the duo redirection process. Server PR in bitwarden/server#3806

The second is making desktop and browser show a handoff message in the connector after the duo authentication. We can't use the same way as web because browser and desktop can't set cookies. Instead, we open a new tab in the duo redirect connector that will set a cookie and redirect the user to the duo connector from there.

## Code changes

- **file.ext:** Description of what was changed and why

## Screenshots

https://github.com/bitwarden/clients/assets/24985544/5c0ac90e-ccf3-456d-a7dd-e3ed337d6be6

https://github.com/bitwarden/clients/assets/24985544/29925c31-70bc-4157-946a-f56d3ae0b0b0

https://github.com/bitwarden/clients/assets/24985544/d22e12d2-e0ee-4987-80d3-9f40a0e3789a

(i know this^ one doesn't show the hand off message, trust me its there 😁 )

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Review by ike-kottlowski — DISMISSED** [review]
LGTM

File: apps/web/src/connectors/duo-redirect.ts [hunk 1]
```diff
@@ -5,24 +5,63 @@ require("./duo-redirect.scss");
...
+    validateUrl = new URL(redirectUrl);
+  } catch (error) {
+    return;
+  }
+  let parts = validateUrl.hostname.split(".");
+  parts = parts.slice(parts.length - 2);
+  const hostname = parts.join(".");
+  if (validateUrl.protocol !== "https:" || hostname !== "duosecurity.com") {
+    return;
+  }
```
Comment (JaredSnider-Bitwarden) [inline apps/web/src/connectors/duo-redirect.ts:60]: Two questions:
(1) Are we worried about duo subdomains like `eu.duosecurity.com`?  Is that a possibility in our implementation? We could make the check more flexible with something like `!hostname.endsWith("duosecurity.com")`?
(2) Should we throw an error along with the return to make debugging potential issues easier?

**Review by JaredSnider-Bitwarden — COMMENTED** [review]
This is looking really good. 2 questions below:

File: apps/web/src/connectors/duo-redirect.ts (hunk 1 above)
Comment (jlf0dev) [reply apps/web/src/connectors/duo-redirect.ts]: Good calls, fixed.

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by ike-kottlowski — APPROVED** [review]

**Review by cagonzalezcs — APPROVED** [review]
We really need to consider pulling that `content-message-handler.ts` file out of team Autofill's scope. The behavior it enables revolves around Auth almost entirely, lol.

**Pull request merged**